### PR TITLE
Various Pulsar extended metadata fixes

### PR DESCRIPTION
**Pull request by jmchilton**
## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy/jobs/command_factory.py [hunk 1]
```diff
@@ -103,10 +105,13 @@ def build_command(
         else:
             commands_builder = CommandsBuilder(externalized_commands)
 
-    for_pulsar = "script_directory" in remote_command_params
-    if not for_pulsar:
+    if externalized:
+        for_pulsar = "pulsar_version" in remote_command_params
+        # Galaxy writes I/O files to outputs, Pulsar uses metadata. metadata seems like
+        # it should be perferred - at least if the directory exists.
```
Comment (mvdbeek) [inline lib/galaxy/jobs/command_factory.py]:
```suggestion
        # it should be preferred - at least if the directory exists.
```

**Review by mvdbeek — APPROVED** [review]

File: lib/galaxy/jobs/command_factory.py [hunk 2]
```diff
@@ -119,6 +124,7 @@ def build_command(
         # xref https://github.com/galaxyproject/galaxy/issues/3289
         commands_builder.prepend_command(PREPARE_DIRS)
 
+    for_pulsar = "script_directory" in remote_command_params
```
Comment (nsoranzo) [inline lib/galaxy/jobs/command_factory.py]: Seems a bit confusing that `for_pulsar` is re-defined here, can the 2 "in" tests give different outcomes?

Comment (jmchilton) [reply lib/galaxy/jobs/command_factory.py]: That is an oversight - I will fix it. Thanks for the catch.

Comment (nsoranzo) [reply lib/galaxy/jobs/command_factory.py]:
```suggestion
```

**Comment by mvdbeek** [comment]
Ready to go @jmchilton ?

**Comment by jmchilton** [comment]
I assume - no way to know for sure.

**Pull request merged**